src/components/Card/Card.js

```diff
@@ -1,30 +1,49 @@
-import React from "react";
+import React, { useState } from "react";
 import { Link } from "react-router-dom";
 
 const Card = ({ page, results, disableLink = true }) => {
-  let display;
+ // Initialize an array to track hover state for each card
+ const [isHovered, setIsHovered] = useState(new Array(results.length).fill(false));
 
-  const cardStyle = {
+ const handleMouseEnter = (index) => {
```

Comment:
I think we can have a more efficient way to implement this but let stick with this and just add it to the tech debt.